### ISSUE TITLE
Custom Formats not evaluated

### ISSUE THREAD

**Issue by zsblevins** [networknt/json-schema-validator#832]
I've created several custom formats but they don't seem to be evaluating:

Sample custom format:

```
public class IpHostPrefixFormat extends AbstractFormat {
    public IpHostPrefixFormat() {
        super("ip_host_prefix", "must be a valid host IP with prefix length");
    }

    public IpHostPrefixFormat(String name, String errorMessageDescription) {
        super(name, errorMessageDescription);
    }

    protected IPAddress convertToAddress(String value) {
        try {
            IPAddressString addressString = new IPAddressString(value);
            return addressString.toAddress();
        } catch(AddressStringException e) {
            return null;
        }
    }

    protected boolean validateAddress(String value, IPAddress address) {
        //Ensure full CIDR supplied
        if (address.getNetworkPrefixLength() == null) {
            return false;
        }
        //Ensure not a network address
        if (address.isPrefixBlock()){
            // Exceptions for /31-32 and /127-128
            if (address.getIPVersion().isIPv4() && address.getNetworkPrefixLength() < 31) {
                return false;
            }
            if (address.getIPVersion().isIPv6() && address.getNetworkPrefixLength() < 127) {
                return false;
            }
        }
        return address.toString().equals(value.toLowerCase());
    }
    @Override
    public boolean matches(String value) {
        IPAddress address = convertToAddress(value);
        if (address == null) {
            return false;
        }
        return validateAddress(value, address);
    }
}
```

I create my JSONSchema as follows:

```
private JsonSchema buildSchema(JsonNode schemaJson) {
        String schemaUri = schemaJson.get("$schema").asText();
        JsonMetaSchema jsonMetaSchema =
                new JsonMetaSchema.Builder(schemaUri).addFormats(formats).build();
        JsonSchemaFactory jsonSchemaFactory =
                new JsonSchemaFactory.Builder().defaultMetaSchemaURI(schemaUri).addMetaSchema(jsonMetaSchema).build();
        return jsonSchemaFactory.getSchema(schemaJson);
    }
```

Where `formats` includes `new IpHostPrefixFormat()` in the list.

If I then test the following schema against the json, it passes validation when it should fail.

```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": false,
  "properties": {
    "cidr": {
      "type": "string",
      "format": "ip_host_prefix"
    }
  },
  "required": ["cidr"],
  "title": "Sample ip schema",
  "type": "object"
}
```

```
{
  "cidr": "bogus"
}
```

I wrote test cases for the `matches` method on every custom formatter and confirmed they work as expected, it just seems like the schema isn't using them. Am I constructing the Meta schema improperly?

Note that standard validators (e.g. string, number, etc) are all working fine in my tests, it's only the custom formats that do not work.

**Comment by zsblevins** [comment]
I'm fairly certain my issue was that I was assuming passing the schema URI would load the right "base" validator and that I was adding additional formats. Looking more closely at the code, that doesn't seem to be the case, but doing:
```
        JsonMetaSchema jsonMetaSchema = new JsonMetaSchema.Builder(schemaUri)
                .idKeyword("$id")
                .addFormats(formats)
                .addKeywords(ValidatorTypeCode.getNonFormatKeywords(SpecVersion.VersionFlag.V7))
                .build();
```
Doesn't seem to help. I can fully recreate the draft 7 code, but will need to dynamically handle multiple drafts.

**Comment by fdutton** [comment]
@zsblevins Take a look at this unit-test and see if this helps. I cannot test your code since I don't have IPAddress and IPAddressString.

https://github.com/networknt/json-schema-validator/blob/master/src/test/java/com/networknt/schema/Issue784Test.java

**Comment by stacywsmith** [comment]
I believe I see the issue:
The custom metaschema is saved into `JsonSchemaFactory.jsonMetaSchemas` [here](https://github.com/networknt/json-schema-validator/blob/master/src/main/java/com/networknt/schema/JsonSchemaFactory.java#L132) with the uri of `http://json-schema.org/draft-07/schema#` as the key.

Later, an attempt to read from this hash map is made [here](https://github.com/networknt/json-schema-validator/blob/master/src/main/java/com/networknt/schema/JsonSchemaFactory.java#L351-L352) with the normalized uri of `https://json-schema.org/draft-07/schema` as the key. That key does not exist, so the standard V7 meta schema is loaded instead. This explains why the built-in formats are working, but the custom formats are not.

**Comment by zsblevins** [comment]
Just wanted to check in on when a release will be created containing this change. While I was able to work around it in my tests, we have a ton of legacy JSON Schemas that we're using that have the non-normalized schema URL format and that we can't safely change without concern that we'd break the existing tooling that interacts with them.

**Comment by stevehu** [comment]
I have released 1.0.86. 

https://github.com/networknt/json-schema-validator/releases/tag/1.0.86
